### PR TITLE
update content from user feedback

### PR DESCRIPTION
**Pull request by simonnineteen90**
https://eaflood.atlassian.net/browse/SFI-1889

### CODE REVIEW

File: app/routes/schemas/debt-year.js [hunk 1]
```diff
@@ -3,15 +3,15 @@ const Joi = require('joi')
 const yearNow = new Date().getFullYear()
 
 module.exports = {
-  'debt-discovered-year': Joi.number().integer().greater(2014).less(1 + yearNow).required()
+  'debt-discovered-year': Joi.number().integer().greater(2020).less(1 + yearNow).required()
```
Comment (marctemp) [inline app/routes/schemas/debt-year.js:6]: This and `app/routes/schemas/enrich-request.js` yea differ -- is this expected?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ffc-pay-request-editor",
-  "version": "1.11.0",
+  "version": "1.12.0",
```
Comment (marctemp) [inline package.json:3]: Need to update `package-lock.json` with this version

File: app/routes/manual-ledger-check.js [hunk 3]
```diff
@@ -40,9 +40,7 @@ module.exports = [{
       query: Joi.object({
         paymentRequestId: Joi.number().required(),
         'ar-value': Joi.number().required(),
-        'ap-value': Joi.number().required(),
-        'ap-percentage': Joi.number().required(),
-        'ar-percentage': Joi.number().required()
+        'ap-value': Joi.number().required()
       }).options({ allowUnknown: true }),
```
Comment (marctemp) [inline app/routes/manual-ledger-check.js:44]: Could, and possibly should, be extracted into its own file à la our payload schemas and imported in to keep the functionality sep. and allow for greater control over expected error messages etc -- I would say this is Out of scope (OOS) for this ticket but could you add this as a tech debt ticket in our backlog?

File: app/routes/schemas/enrich-request.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ const {
 module.exports = Joi.object({
   day: Joi.number().integer().min(1).max(31).required(),
   month: Joi.number().integer().min(1).max(12).required(),
-  year: Joi.number().integer().min(2020).max(9999).required(),
+  year: Joi.number().integer().min(2021).max(9999).required(),
   'debt-type': Joi.string().valid(ADMINISTRATIVE, IRREGULAR).required(),
   'invoice-number': Joi.string().required()
 })
```
Comment (marctemp) [inline app/routes/schemas/enrich-request.js:14]: This should follow the same pattern as implemented throughout this repo, i.e. define individual schemas for each component (day, month, year, debt-type and invoice-number) with customer defined error codes and import them into page routes as individual use-cases, i.e. `enrich-request`, `capture-debt`, etc (add a tech debt ticket if 1 already doesn't exist)

Also, we should standardise which Joi functions we use, i.e. here we're using `min`, `max` but else where for ints we use `greater` and `less` -- which is more intuitive? Again another tech debt ticket

File: app/routes/models/enrich-request.js [hunk 5]
```diff
@@ -99,7 +99,7 @@ function ViewModel (payload, error) {
         }
```
Comment (marctemp) [inline app/routes/models/enrich-request.js:99]: On line 84, `checked: getObjectKeyEquals(payload, 'debtType', IRREGULAR)` needs changed to `checked: getObjectKeyEquals(payload, 'debt-type', IRREGULAR)` and similar for line 89

File: app/views/capture.njk [hunk 6]
```diff
@@ -45,13 +72,13 @@
...
                 <th scope="col" class="govuk-table__header"></th>
               </tr>
             </thead>
             <tbody class="govuk-table__body">
               {% for data in captureData %}
-                <tr class="govuk-table__row">
+                <tr class="govuk-table__row capture-datasets">
                   <td scope="col" class="govuk-table__cell govuk-body-s">{{ data.schemes.name }}</td>
                   <td scope="col" class="govuk-table__cell govuk-body-s">{{ data.frn }}</td>
                   <td scope="col" class="govuk-table__cell govuk-body-s">{% if data.reference %}{{ data.reference }}{%else%}Manual enrichment{%endif%}</td>
```
Comment (marctemp) [inline app/views/capture.njk:84]: On my local testing using the `payload` given in the `README`, I am seeing:

-  for the col `Agreement number` I am seeing `Manual enrichment` despite the `payload` having an `Agreement number`
- for `Net value` a value of `£0.00`

How to recreate:

- send 1 `README payload` to `ffc-pay-debt-data`
- enrich it via `/enrich`
- view `/capture`

Will attach a screenshot to the ticker in Jira

File: test/integration/narrow/routes/enrich-request.test.js [hunk 7]
```diff
@@ -37,8 +37,9 @@ describe('Enrich request test', () => {
     marketingYear: 2022,
     currency: 'GBP',
     schedule: 'M12',
-    dueDate: '2021-08-15',
+    dueDate: '2021-10-25',
     value: 15000,
+    received: '2021-10-25',
```
Comment (marctemp) [inline test/integration/narrow/routes/enrich-request.test.js:42]: Can you add some tests around making sure `daysWaiting` is displayed correctly?

File: app/debt/get-debts.js [hunk 8]
```diff
@@ -23,7 +23,8 @@ const getDebts = async (includeAttached = false) => {
       'createdBy',
       'attachedDate',
       'paymentRequestId'
-    ]
+    ],
+    order: [['createdDate', 'DESC']]
```
Comment (marctemp) [inline app/debt/get-debts.js:27]: Can you add tests in `test/integration/narrow/routes/capture.test.js` to ensure this ordering is correctly displayed?

File: app/payment-request/get-payment-request.js [hunk 9]
```diff
@@ -25,8 +25,11 @@ const getPaymentRequest = async (categoryId = 1) => {
       'received',
       'receivedFormatted',
       'ledger',
+      'marketingYear',
+      'daysWaiting',
       'netValue'
-    ]
+    ],
+    order: [['received']]
```
Comment (marctemp) [inline app/payment-request/get-payment-request.js:32]: Can you add tests in `test/integration/local/payment-request/get-payment-request.test.js` to ensure this ordering is correctly displayed, both ordering and new fields?

File: app/views/capture.njk (hunk 6 above)
Comment (marctemp) [reply app/views/capture.njk:84]: Also, use whitespace to help break up this `if-else` similar to lines 90 - 94

File: app/views/enrich.njk [hunk 10]
```diff
@@ -33,25 +33,23 @@
...
-                <th scope="col" class="govuk-table__header govuk-table__header--numeric govuk-body-s">Arrived</th>
+                <th scope="col" class="govuk-table__header govuk-table__header--numeric govuk-body-s">Days waiting</th>
                 <th scope="col" class="govuk-table__header"></th>
               </tr>
             </thead>
             <tbody class="govuk-table__body">
               {% for data in enrichData %}
                 <tr class="govuk-table__row">
                   <td scope="col" class="govuk-table__cell govuk-body-s">{{ data.schemes.name }}</td>
+                  <td scope="col" class="govuk-table__cell govuk-body-s">{{ data.marketingYear }}</td>
```
Comment (marctemp) [inline app/views/enrich.njk:48]: Should marketing year be tagged as `govuk-table__cell--numeric`, also <-- what does that do?

File: test/integration/narrow/routes/enrich-request.test.js [hunk 11]
```diff
@@ -271,7 +272,7 @@ describe('Enrich request test', () => {
       expect(debtDataRow[0].schemeId).toBe(1)
```
Comment (marctemp) [inline test/integration/narrow/routes/enrich-request.test.js:272]: We have a constant module which defined scheme ids -- can you replace these hardcoded values with them?

File: test/integration/narrow/routes/manual-ledger-check.test.js [hunk 12]
```diff
@@ -40,6 +40,7 @@ describe('Manual-ledger-check tests', () => {
 
   afterEach(async () => {
     await server.stop()
+    jest.clearAllMocks()
```
Comment (marctemp) [inline test/integration/narrow/routes/manual-ledger-check.test.js:43]: Whata method <3

File: app/views/macros/payment-request-header.njk [hunk 13]
```diff
@@ -4,23 +4,29 @@
...
-            <th scope="col" class="govuk-table__header govuk-table__header--numeric govuk-body-s">Ledger</th>
+            {% if not paymentRequestData.manualLedgerChecks%}
+              <th scope="col" class="govuk-table__header govuk-table__header--numeric govuk-body-s">Ledger</th>
+            {% endif %}
           </tr>
         </thead>
         <tbody class="govuk-table__body">
           <tr class="govuk-table__row">
             <td class="govuk-table__cell govuk-body-s">{{paymentRequestData.schemes.name}}</td>
+            <td class="govuk-table__cell govuk-body-s">{{paymentRequestData.marketingYear}}</td>
```
Comment (marctemp) [inline app/views/macros/payment-request-header.njk:21]: Same comment on marketing year

File: package.json (hunk 2 above)
Comment (simonnineteen90) [reply package.json:3]: updated to 1.12.0

File: app/routes/models/enrich-request.js (hunk 5 above)
Comment (simonnineteen90) [reply app/routes/models/enrich-request.js:99]: updated 'debtType' to 'debt-type'

File: app/views/capture.njk [hunk 14]
```diff
@@ -89,4 +116,27 @@
...
+        }
+      }
+    } else {
+      tableRows.forEach(x => x.removeAttribute('style'))
+    }
+  }
+
+  toggleUnattachedForm.forEach(x => x.addEventListener('change', toggleUnattachedData))
+ 
+  </script>
```
[the remark this answers is not in the document]
Comment (simonnineteen90) [reply app/views/capture.njk:141]: updated

File: app/views/enrich.njk (hunk 10 above)
Comment (simonnineteen90) [reply app/views/enrich.njk:48]: i think it right aligns the cell data for easy comparision between columns according to the docs [here](https://design-system.service.gov.uk/components/table/#numbers-in-a-table)

Therefore dont think we need it as right align makes it look odd.

File: app/views/macros/payment-request-header.njk (hunk 13 above)
Comment (simonnineteen90) [reply app/views/macros/payment-request-header.njk:21]: Dont think this is required as above comment.

File: app/views/capture.njk (hunk 6 above)
Comment (simonnineteen90) [reply app/views/capture.njk:84]: Resolved this with Steve by updating the `get-debts` query to filter out the applications which have a `reference` of `Manual enrichment`

File: test/integration/narrow/routes/enrich-request.test.js (hunk 7 above)
Comment (simonnineteen90) [reply test/integration/narrow/routes/enrich-request.test.js:42]: added tests to `get-payment-request.test.js`

File: app/debt/get-debts.js (hunk 8 above)
Comment (simonnineteen90) [reply app/debt/get-debts.js:27]: added test to get-debts-test

File: app/quality-check/get-quality-checks.js [hunk 15]
```diff
@@ -36,7 +38,7 @@ const getQualityChecks = async () => {
       status: PENDING
     }
   })
-  return qualityChecks.map(x => x.get({ plain: true }))
+  return getChangedQualityChecks(qualityChecks)
```
[the remark this answers is not in the document]
Comment (simonnineteen90) [reply app/quality-check/get-quality-checks.js:41]: Covered all action points however noticed there are missing tests and suggest we need to raise a tech debt ticket to cover these.

File: app/views/manual-ledger-check.njk [hunk 16]
```diff
@@ -119,26 +139,17 @@
       return valueInPence / 100
```
[the remark this answers is not in the document]
Comment (simonnineteen90) [reply app/views/manual-ledger-check.njk:139]: dont think i can import node module into njk file as frontend

File: test/integration/narrow/routes/capture.test.js [hunk 17]
```diff
@@ -70,7 +70,6 @@ describe('Capture test', () => {
       expect(response.request.response.source.template).toBe('capture')
     })
   })
-
```
Comment (marctemp) [inline test/integration/narrow/routes/capture.test.js:73]: Would add back in

File: test/unit/quality-check/get-changed-quality-checks.test.js [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+        frn: 1234567890,
+        paymentRequest: {
+          paymentRequestId: 5
+        }
+      }]
+    const changedQualityCheck = await getChangedQualityChecks(qualityChecks)
+    console.log(changedQualityCheck)
+    expect(changedQualityCheck[0].hasDismissed).toBe('No')
+  })
+  test('should return hasDismissed as no with paymentRequestId of 4 and "active" is true', async () => {
```
Comment (marctemp) [inline test/unit/quality-check/get-changed-quality-checks.test.js:64]: This test isn't testing that "active" is true

File: test/unit/quality-check/get-changed-quality-checks.test.js [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+  test('should return hasDismissed as yes with paymentRequestId of 1', async () => {
+    const qualityChecks = [
+      {
+        frn: 1234567890,
+        paymentRequest: {
+          paymentRequestId: 1
+        }
+      }]
+    const changedQualityCheck = await getChangedQualityChecks(qualityChecks)
+    console.log(changedQualityCheck)
```
Comment (marctemp) [inline test/unit/quality-check/get-changed-quality-checks.test.js:49]: Couple of logs in this file

File: test/integration/narrow/routes/enrich-request.test.js (hunk 11 above)
Comment (simonnineteen90) [reply test/integration/narrow/routes/enrich-request.test.js]: updated

File: app/payment-request/get-payment-request.js (hunk 9 above)
Comment (simonnineteen90) [reply app/payment-request/get-payment-request.js:32]: added test

File: test/integration/narrow/routes/capture.test.js (hunk 17 above)
Comment (simonnineteen90) [reply test/integration/narrow/routes/capture.test.js:73]: added

File: test/unit/quality-check/get-changed-quality-checks.test.js [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+    expect(changedQualityCheck[0].hasDismissed).toBe('Yes')
+  })
+  test('should return hasDismissed as no with paymentRequestId of 5', async () => {
```
[the remark this answers is not in the document]
Comment (simonnineteen90) [reply test/unit/quality-check/get-changed-quality-checks.test.js:52]: added

File: test/unit/quality-check/get-changed-quality-checks.test.js [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+    paymentRequest = {
+      paymentRequestId: 1,
+      schemeId: SCHEME_ID_SFI,
+      frn: 1234567890,
+      categoryId: 2
+    }
+
+    await db.scheme.create(scheme)
+    await db.paymentRequest.create(paymentRequest)
+    await db.paymentRequest.create({ ...paymentRequest, paymentRequestId: 2 })
```
[the remark this answers is not in the document]
Comment (simonnineteen90) [reply test/unit/quality-check/get-changed-quality-checks.test.js:26]: Steve takes the credit haha

File: test/integration/local/payment-request/get-payment-request.test.js [hunk 22]
```diff
@@ -57,4 +57,23 @@ describe('Get payment request test', () => {
     expect(paymentRequests[0].receivedFormatted).not.toBe(null)
     expect(paymentRequests[0].receivedFormatted).toBe(paymentRequests[0].received.toLocaleDateString('en-GB', { year: 'numeric', month: '2-digit', day: '2-digit' }))
   })
+
+  test('daysWaiting virtual type should return the number of days since the "recieved" date', async () => {
+    paymentRequest.received = '01-01-2022'
+    await resetData()
+    await db.paymentRequest.create(paymentRequest)
+    const daysWaiting = (Math.round((Date.now() - new Date(paymentRequest.received)) / (1000 * 60 * 60 * 24)))
```
[the remark this answers is not in the document]
Comment (simonnineteen90) [reply test/integration/local/payment-request/get-payment-request.test.js:65]: implement mocking for Date and updated test.

File: test/unit/quality-check/get-changed-quality-checks.test.js (hunk 18 above)
Comment (simonnineteen90) [reply test/unit/quality-check/get-changed-quality-checks.test.js:64]: I think this okay as line 32 sets `active` to `true` for a manual ledger record with a `paymentRequestID` of 4

File: app/routes/schemas/enrich-request.js (hunk 4 above)
Comment (simonnineteen90) [reply app/routes/schemas/enrich-request.js:14]: added tickets to backlog

File: test/unit/quality-check/get-changed-quality-checks.test.js (hunk 18 above)
Comment (marctemp) [reply test/unit/quality-check/get-changed-quality-checks.test.js:64]: Aaaaahhh I gotcha now -- yup, looking up at the `beforeEach` helped.

Hhmmmmmm I would reword these test cases so tht they reference the state that the payment record is in

Instead of `should return hasDismissed as yes with paymentRequestId of 1'` because this test would fail if you `beforeEach` didn't inc. `paymentRequestId` 1 ya'know? It isn't dependent on that.

What it is dependent on is the fact that `paymentRequestId` 1 exists within `paymentRequest` table and that matching record exists within `manualLedgerPaymentRequest` with `active` equal `false`

So I'd consider something like: `should return hasDismissed as "Yes" (being picky over casing and exact return value heheh) when paymentRequest exists and matching manualLedgerPaymentRequest exists with active set to false`

This name doesn't care what the `paymentRequestId` is (becuz we make tht up in our mock data) but it gives context to the state of this particular record and particular test scenario -- which gives way to asking more questions like, "What about when there isn't a matching `paymentRequest` in `manualLedgerPaymentRequest`, e.g. `paymentRequestId` 2 in your `beforeEach`?"

Comment (simonnineteen90) [reply test/unit/quality-check/get-changed-quality-checks.test.js]: updated test names as suggested.
Considered if additional test needed to cover above point of _What about when there isn't a matching paymentRequest in manualLedgerPaymentRequest, e.g. paymentRequestId 2_ 
However i think this is covered by second test on line 52 where there is a record in `paymentRequest` table but no record in `manualLedgerRequest` table but for `paymentRequestId` 5. Therefore no additional tests added

File: test/integration/narrow/routes/capture-debt.test.js [hunk 23]
```diff
@@ -1751,4 +1751,23 @@ describe('capture-debt route', () => {
     expect(debtData.schemeId).toBe(1)
     expect(debtData.frn).toBe(VALID_PAYLOAD.frn)
   })
+  test('POST /capture-debt with future date returns 400 ', async () => {
```
[the remark this answers is not in the document]
Comment (simonnineteen90) [reply test/integration/narrow/routes/capture-debt.test.js]: reworded name of this test and added additional test to cover a future date scenario.

File: app/views/home.njk [hunk 24]
```diff
@@ -38,7 +38,7 @@
     {% if auth.isLedgerUser %}
     <div class="govuk-cards govuk-cards--2">
         <div class="govuk-cards--2__item govuk-card--dashboard">
-          <h2 class="govuk-heading-m govuk-!-margin-bottom-2">Awaiting Manual Ledger Assignment</h2>
+          <h2 class="govuk-heading-m govuk-!-margin-bottom-2">Awaiting ledger assignment</h2>
```
Comment (marctemp) [inline app/views/home.njk:41]: Final comment from my perspective:

The link under this reads `View Manual Ledger Assignment` should this be `View awaiting ledger assignment` to be consistent with the title and the other links on the page?

Comment (simonnineteen90) [reply app/views/home.njk:41]: agree with this. updated as suggested

**Review by marctemp — APPROVED** [review]
Approved -- good job
